src/elm/Eos/EosError.elm

```diff
@@ -0,0 +1,50 @@
+module Eos.EosError exposing (prepareErrorMessage)
+
+import Json.Decode as Decode exposing (at, decodeString, field, list, string)
+import Session.Shared exposing (Translators)
+
+
+{-| Extracts failure description from the first error message received from the EOS:
+
+    extractFailure "assertion failure with message: Can't vote on already verified claim"
+        == "Can't vote on already verified claim"
+
+The result is a string which can be translated and showed to the user.
+
+-}
+extractFailure : String -> String
+extractFailure json =
+    let
+        eosErrorMessages =
+            decodeString decodeErrorDetails json
+    in
+    case eosErrorMessages of
+        Ok (firstMessage :: _) ->
+            case String.split ": " firstMessage of
+                _ :: msg :: [] ->
+                    msg
+
+                _ ->
+                    -- Keep whole message (may be verbose, with no translation, but better than nothing)
+                    firstMessage
+
+        _ ->
+            "error.unknown"
+
+
+prepareErrorMessage : Translators -> Maybe String -> String
+prepareErrorMessage { t } eosErrorString =
```

Comment:
I'm not 100% sure about this name, maybe `translateErrorMessage` or `parseErrorMessage`. The word `prepare` seems out of place, idk

Reply:
Agree! Let's go with `parseErrorMessage`.